app/src/androidTest/java/ch/epfl/sdp/peakar/TestMenuBarActivityTest.java

```diff
@@ -0,0 +1,62 @@
+package ch.epfl.sdp.peakar;
+
+import android.Manifest;
+
+import androidx.test.espresso.intent.Intents;
+import androidx.test.ext.junit.rules.ActivityScenarioRule;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import androidx.test.rule.GrantPermissionRule;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(AndroidJUnit4.class)
+public class TestMenuBarActivityTest {
+    @Rule
+    public ActivityScenarioRule<TestMenuBarActivity> testRule = new ActivityScenarioRule<>(TestMenuBarActivity.class);
+
+    /* Create Intent */
+    @Before
+    public void setup(){
+        Intents.init();
+    }
+
+    /* Release Intent */
+    @After
+    public void cleanUp(){
+        Intents.release();
+    }
+
+    /* Test that menu bars settings icon works as intended */
+    @Test
+    public void TestMenuBarSettings(){
+        MenuBarTestHelper.TestClickableIconButton(R.id.menu_bar_settings);
+    }
+
+    /* Test that menu bars gallery icon works as intended */
+    @Test
+    public void TestMenuBarGallery(){
+        MenuBarTestHelper.TestClickableIconButton(R.id.menu_bar_gallery);
+    }
+
+    /* Test that menu bars camera icon works as intended */
+    /*@Test
+    public void TestMenuBarCamera(){
+        MenuBarTestHelper.TestClickableIconButton(R.id.menu_bar_camera);
```

Comment:
It seems like this is a problem with espresso. The button is clicked but the intent is never checked. Take a look at https://stackoverflow.com/a/39379856 maybe you could try creating an own IdlingResources.

Reply:
After discussion removed test

Reply:
After some debugging, I found, that something runs in the background preventing espresso to continue testing. This is probably due to the CameraUiView which checks for the POIPoints. However, since they are never computed they will stay empty. You can either compute the POIs from your test or simply move the test file to the general package (where I think it should belong) s.t. the POIs are already computed from the InitActivityTest.